### PR TITLE
clean the options with information for cylc rose after dumping thier …

### PR DESCRIPTION
**Pull request by wxtim**
…content in rose-suite-cylc-install.conf

Partner of https://github.com/cylc/cylc-flow/pull/6060

**Check List**

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` (and `conda-environment.yml` if present).
- [x] Tests are included (or explain why tests are not needed).
- [x] `CHANGES.md` entry included if this is a change that can affect users
- [x] [Cylc-Doc](https://github.com/cylc/cylc-doc) pull request opened if required at cylc/cylc-doc/pull/XXXX.
- [x] If this is a bug fix, PR should be raised against the relevant `?.?.x` branch.

### CODE REVIEW

File: tests/unit/test_functional_post_install.py [hunk 1]
```diff
@@ -355,3 +355,20 @@ def test_cylc_no_rose(tmp_path):
...
+    monkeypatch.setattr(f'{path}copy_config_file', lambda *_, **__: False)
+    monkeypatch.setattr(f'{path}rose_config_exists', lambda *_, **__: True)
+
+    opts = SimpleNamespace(
+        rose_template_vars=['gee', 'whiz'],
+        defines=['foo', 'bar'],
+        opt_conf_keys=['thomas', 'percy'])
+
+    post_install('', '', opts)
+    assert any(opts.__dict__.values()) is False
```
Comment (oliver-sanders) [inline tests/unit/test_functional_post_install.py:374]: For a functional test, how about something installs and then reinstalls a workflow using the same `opts` object e.g:

```python
def test_something(cylc_install_cl):
    opts = SimpleNamespace(
        rose_template_vars=['gee', 'whiz'],
        defines=['foo', 'bar'],
        opt_conf_keys=['thomas', 'percy']
    )

    # install
    run_dir = cylc_install_cli(..., opts)

    # check installed rose config
    RoseConfigTree(run_dir).get(('template variables', 'gee')) == 'whatever'

    # reinstall using same opts object (just like cylc-flow does)
    run_dir = cylc_install_cli(..., opts)

     # check the installed rose config again
    RoseConfigTree(run_dir).get(('template variables', 'gee')) == 'whatever'
```

File: cylc/rose/entry_points.py [hunk 2]
```diff
@@ -71,6 +71,12 @@ def post_install(srcdir: Path, rundir: str, opts: 'Values') -> bool:
     if config_node:
         dump_rose_log(rundir=_rundir, node=config_node)
 
+    # Clear opts before returning; they have now been stored in
+    # rose-suite-cylc-install.conf
```
Comment (oliver-sanders) [inline cylc/rose/entry_points.py:75]: Please link to issue.

Comment (wxtim) [reply cylc/rose/entry_points.py:75]:
```suggestion
    # Clear opts before returning; they have now been stored in
    # rose-suite-cylc-install.conf
    # https://github.com/cylc/cylc-flow/issues/6058
```

File: tests/unit/test_functional_post_install.py (hunk 1 above)
Comment (wxtim) [reply tests/unit/test_functional_post_install.py:374]: >     # reinstall using same opts object (just like cylc-flow does)

Sorry, but Cylc Flow doesn't reinstall with the same opts object as Cylc install - do you mean play?

Comment (wxtim) [reply tests/unit/test_functional_post_install.py:374]: I don't understand how this is supposed to demonstate the broken state before my changes. `cylc reinstall` interacts with the Opts and the source directory (the validate step in VR does the interaction with the run dir). In this bug the problem is the interaction between the opts and the run directory when `cylc play` is run. I'm happy to do _that_ as a functional test.

Comment (wxtim) [reply tests/unit/test_functional_post_install.py:374]: I don't think that this makes sense: the problem was with the content of the run directory interacting with `cylc play`, not the source directory. I'll translate my manual test into one of these.

**Pull request closed without merging**